packages/@aws-cdk/aws-sagemaker/lib/model.ts

```diff
@@ -0,0 +1,405 @@
+   *
+   * @default - none
+   */
+  readonly containerHostname?: string;
+
+  /**
+   * S3 path to the model artifacts.
+   *
+   * @default - none
+   */
+  readonly modelData?: ModelData;
+}
+
+/**
+ * Construction properties for a SageMaker Model.
+ */
+export interface ModelProps {
+
+  /**
+   * The IAM role that the Amazon SageMaker service assumes.
+   *
+   * @default - a new IAM role will be created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Name of the SageMaker Model.
+   *
+   * @default - AWS CloudFormation generates a unique physical ID and uses that ID for the model's
+   * name.
+   */
+  readonly modelName?: string;
+
+  /**
+   * The VPC to deploy model containers to.
+   *
+   * @default - none
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * The VPC subnets to use when deploying model containers.
+   *
+   * @default - none
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * The security groups to associate to the Model. If no security groups are provided and 'vpc' is
+   * configured, one security group will be created automatically.
+   *
+   * @default - A security group will be automatically created if 'vpc' is supplied
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * Specifies the container definitions for this model, consisting of either a single primary
+   * container or an inference pipeline of multiple containers.
+   *
+   * @default - none
+   */
+  readonly containers?: ContainerDefinition[];
+
+  /**
+   * Whether to allow the SageMaker Model to send all network traffic
+   *
+   * If set to false, you must individually add traffic rules to allow the
+   * SageMaker Model to connect to network targets.
+   *
+   * Only used if 'vpc' is supplied.
+   *
+   * @default true
+   */
+  readonly allowAllOutbound?: boolean;
+}
+
+/**
+ * Defines a SageMaker Model.
+ */
+export class Model extends ModelBase {
+  /**
+   * Imports a Model defined either outside the CDK or in a different CDK stack.
+   * @param scope the Construct scope.
+   * @param id the resource id.
+   * @param modelName the name of the model.
+   */
+  public static fromModelName(scope: Construct, id: string, modelName: string): IModel {
+    return Model.fromModelAttributes(scope, id, { modelName });
+  }
+
+  /**
+   * Imports a Model defined either outside the CDK or in a different CDK stack.
+   * @param scope the Construct scope.
+   * @param id the resource id.
+   * @param attrs the attributes of the model to import.
+   */
+  public static fromModelAttributes(scope: Construct, id: string, attrs: ModelAttributes): IModel {
+    const modelName = attrs.modelName;
+    const role = attrs.role;
+
+    class Import extends ModelBase {
+      public readonly modelName = modelName;
+      public readonly role = role;
+      public readonly grantPrincipal: iam.IPrincipal;
+      public readonly modelArn: string;
+
+      constructor(s: Construct, i: string) {
+        super(s, i);
```

Comment:
```suggestion
        super(s, i, {
          environmentFromArn: modelArn,
        });
```

Reply:
this gets the environment from the arn, so that we serve imports from different envs.